src/stcal/jump/twopoint_difference.py

```diff
@@ -153,24 +153,44 @@ def find_crs(
     num_flagged_grps = 0
     # determine the number of groups with all pixels set to DO_NOT_USE
     ngrps = dat.shape[1]
+    max_flagged_grps = 0
+    total_flagged_grps = 0
     for integ in range(nints):
+        num_flagged_grps = 0
         for grp in range(dat.shape[1]):
             if np.all(np.bitwise_and(gdq[integ, grp, :, :], dnu_flag)):
                 num_flagged_grps += 1
+            if num_flagged_grps > max_flagged_grps:
```

Comment:
The adjustment of `max_flagged_grps` can be done after the `grp` loop.  Doing this check for every loop over `grp` is inefficient.